core/src/main/java/org/apache/druid/storage/local/LocalFileStorageConnector.java

```diff
@@ -107,6 +111,23 @@ public void deleteRecursively(String dirName) throws IOException
     FileUtils.deleteDirectory(fileWithBasePath(dirName));
   }
 
+  @Override
+  public List<String> listDir(String dirName)
+  {
+    File directory = fileWithBasePath(dirName);
+    if (!directory.exists()) {
+      throw new IAE("No directory exists on path [%s]", dirName);
+    }
+    if (!directory.isDirectory()) {
+      throw new IAE("Cannot list contents of [%s] since it is not a directory", dirName);
+    }
+    File[] files = directory.listFiles();
+    if (files == null) {
```

Comment:
I think empty files should not return an exception but that can be done as part of a follow up PR